libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1830,6 +1830,16 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 
 	if (exprType->category() == Type::Category::Contract)
 	{
+		// Warn about using address members on contracts
+		for (auto const& addressMember: IntegerType(160, IntegerType::Modifier::Address).nativeMembers(nullptr))
+			if (addressMember.name == memberName && *annotation.type == *addressMember.type)
+				m_errorReporter.warning(
```

Comment:
Actually shouldn't we turn this into an error with 0.5.0?

Reply:
We should turn it into an assertion with 0.5.0